### PR TITLE
add REST implementation for GetSyncSubcommitteeIndex

### PR DESCRIPTION
**Pull request by dB2510**
**What type of PR is this?**
Feature

**Related issue**
#11580 

**What does this PR do? Why is it needed?**
It adds REST API implementation for `GetSyncSubcommitteeIndex`.

### CODE REVIEW

File: validator/client/beacon-api/sync_committee.go [hunk 1]
```diff
@@ -93,6 +95,37 @@ func (c *beaconApiValidatorClient) getSyncCommitteeContribution(
...
+
+	var indices []primitives.CommitteeIndex
+	for _, idx := range syncDuty.ValidatorSyncCommitteeIndices {
+		syncCommIdx, err := strconv.ParseUint(idx, 10, 64)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse validator sync committee index")
+		}
+
+		indices = append(indices, primitives.CommitteeIndex(syncCommIdx))
+	}
```
Comment (nalepae) [inline validator/client/beacon-api/sync_committee.go:124]:
```suggestion
	indices := make([]primitives.CommitteeIndex, len(syncDuty.ValidatorSyncCommitteeIndices))

	for i, idx := range syncDuty.ValidatorSyncCommitteeIndices {
		syncCommIdx, err := strconv.ParseUint(idx, 10, 64)
		if err != nil {
			return nil, errors.Wrap(err, "failed to parse validator sync committee index")
		}

		indices[i] = primitives.CommitteeIndex(syncCommIdx)
	}
```

Comment (dB2510) [reply validator/client/beacon-api/sync_committee.go:124]: Hi @nalepae !
Slice initialisation with make is recommended when the length of slice is constant or already known. In our case it length of `indices` depends on length  `syncDuty.ValidatorSyncCommitteeIndices` slice. If `syncDuty.ValidatorSyncCommitteeIndices` slice is empty/nil then it doesn't make sense to pre-allocate indices slice.

File: validator/client/beacon-api/sync_committee.go [hunk 2]
```diff
@@ -7,6 +7,8 @@ import (
 	"net/url"
 	"strconv"
 
+	"github.com/prysmaticlabs/prysm/v3/time/slots"
+
```
Comment (rkapka) [inline validator/client/beacon-api/sync_committee.go:11]:
```suggestion
```

File: validator/client/beacon-api/beacon_api_validator_client.go [hunk 3]
```diff
@@ -76,25 +76,15 @@ func (c *beaconApiValidatorClient) GetFeeRecipientByPubKey(ctx context.Context,
 }
 
 func (c *beaconApiValidatorClient) GetSyncCommitteeContribution(ctx context.Context, in *ethpb.SyncCommitteeContributionRequest) (*ethpb.SyncCommitteeContribution, error) {
-	if c.fallbackClient != nil {
-		return c.fallbackClient.GetSyncCommitteeContribution(ctx, in)
-	}
-
-	// TODO: Implement me
-	panic("beaconApiValidatorClient.GetSyncCommitteeContribution is not implemented. To use a fallback client, create this validator with NewBeaconApiValidatorClientWithFallback instead.")
+	return c.getSyncCommitteeContribution(ctx, in)
```
Comment (rkapka) [inline validator/client/beacon-api/beacon_api_validator_client.go:79]: Nice catch

File: validator/client/beacon-api/sync_committee.go [hunk 4]
```diff
@@ -93,6 +95,37 @@ func (c *beaconApiValidatorClient) getSyncCommitteeContribution(
...
+	}
+
+	// First sync duty is required since we requested sync duties for one validator index.
+	syncDuty := syncDuties[0]
+
+	var indices []primitives.CommitteeIndex
+	for _, idx := range syncDuty.ValidatorSyncCommitteeIndices {
+		syncCommIdx, err := strconv.ParseUint(idx, 10, 64)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse validator sync committee index")
```
Comment (rkapka) [inline validator/client/beacon-api/sync_committee.go:120]:
```suggestion
			return nil, errors.Wrapf(err, "failed to parse validator sync committee index %s", idx)
```
I think that's how we usually return parse errors

File: validator/client/beacon-api/sync_committee_test.go [hunk 5]
```diff
@@ -7,6 +7,8 @@ import (
 	"fmt"
 	"testing"
 
+	"github.com/prysmaticlabs/prysm/v3/time/slots"
+
```
Comment (rkapka) [inline validator/client/beacon-api/sync_committee_test.go:11]:
```suggestion
```

**Review by rkapka — APPROVED** [review]

**Pull request merged**